dace/codegen/targets/fpga.py

```diff
@@ -284,6 +285,22 @@ def fpga_ptr(name: str,
     return name
 
 
+def unqualify_fpga_array_name(sdfg: dace.SDFG, arr_name: str):
+    '''
+    Returns the unqualified array name if it refers to an array interface.
+    Otherwise return it as it is.
+    :param name: array name to unqualify
+    '''
+
+    unqualified = re.sub('_in$|_out$', '', arr_name)
```

Comment:
I would prefix this with `if arr_name not in sdfg.arrays and (arr_name.endswith('_in') or arr_name.endswith('out'))`. Regexps can be quite slow(!)